frontend/rust-lib/flowy-database2/src/services/group/controller_impls/select_option_controller/single_select_controller.rs

```diff
@@ -111,6 +111,22 @@ impl GroupCustomize for SingleSelectGroupController {
 
     Ok((Some(new_type_option.into()), Some(inserted_group_pb)))
   }
+
+  fn delete_group_custom(&mut self, group_id: &str) -> FlowyResult<Option<TypeOptionData>> {
```

Comment:
```rust
fn delete_group_custom(&mut self, group_id: &str) -> FlowyResult<Option<TypeOptionData>> {
    if let Some(option_index) = self.type_option.options.iter().position(|option| option.id == group_id) {
        // Clone the type_option only if the group is found
        let mut new_type_option = self.type_option.clone();
        new_type_option.options.remove(option_index);
        Ok(Some(new_type_option))
    } else {
        // Return None if no matching group is found
        Ok(None)
    }
}

```

Also, you can refactor `delete_option` that accepts a &option instead of the option